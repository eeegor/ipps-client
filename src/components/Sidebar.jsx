import React from 'react';
import { Button, Title, Text, Filter } from '.';
import './Sidebar.scss';

export const Sidebar = props => {
  const {
    isAuth,
    onShowAuthForm,
    onSetFilterField,
    onApplyFilter,
    onLogout,
    filterData
  } = props;
  return (
    <div className="sidebar">
      {!isAuth && (
        <div className="info">You must be logged in to see this page :)</div>
      )}
      {isAuth && (
        <div className="sidebar__container">
          <Title text="IPPS Patient Data" />
          <Text text="Provider Summary for the Top 100 Diagnosis-Related Groups" />

          <hr />

          <div className="sidebar__actions">
            <Button
              className="button--logout button--outline"
              onClick={e => onLogout(e)}
            >
              Logout
            </Button>
          </div>

          <hr />
          {/* <div className="switch">
            <div className="switch__title">
              <h3>View Mode</h3>
            </div>
            <div className="switch__cells grid--2">
              <div className="switch__cell">Table</div>
              <div className="switch__cell">Chart</div>
            </div>
          </div>

          <div className="switch">
            <div className="switch__title">
              <h3>Data</h3>
            </div>
            <div className="switch__cells grid--2">
              <div className="switch__cell">Load All Data</div>
              <div className="switch__cell">Select Columns</div>
            </div>
          </div> */}
          <Filter
            filterData={filterData}
            submitLabel="Apply filter"
            onChange={(field, event) =>
              onSetFilterField(field, event.target.value)
            }
            onApplyFilter={event => onApplyFilter(event)}
            onSubmit={event => onApplyFilter(event)}
          />
        </div>
      )}
    </div>
  );
};
