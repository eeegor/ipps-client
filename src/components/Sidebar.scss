@import '../variables';

.sidebar {
  width: 100%;
  flex: 1 0 auto;
  background: $color--primary;
  color: #fff;
  overflow-y: scroll;
  -webkit-overflow-scrolling: touch;
  position: relative;
  z-index: 9100;

  &__container {
    padding: 1rem;
    padding-bottom: 10rem;
    max-width: 400px;
    margin: 0 auto;
  }

  @media (min-width: $size--tablet) {
    border-left: 1px solid #ddd;
    max-width: 360px;
  }
  @media (min-width: $size--desktop) {
    max-width: 400px;
  }

  .title {
    max-width: 10rem;
    margin-top: 6rem;
    margin-bottom: 4rem;
  }

  .text {
    max-width: 16rem;
  }

  hr {
    border-color: rgba(#fff, 0.3);
  }

  .button--outline {
    color: #fff;
    border-color: rgba(#fff, 0.5);
    padding: 0 0.5rem;
    margin: 0.5rem 0;
    height: 3rem;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 0.8rem;

    &:hover {
      background: rgba(#fff, 0.3);
    }
  }

  .button--logout {
    &:hover {
      border-color: rgba($color--danger, 0.5);
      background: $color--danger;
      color: white;
    }
  }
}

.sidebar__actions {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 1rem;
}

.sidebar-controls {
  .apply-filter {
    display: none;
    position: fixed;
    background: linear-gradient(
      180deg,
      darken($color--primary, 10%),
      rgba($color--primary, 0)
    );
    top: 0;
    left: 0;
    right: 0;
    padding: 0.75rem;
    padding-bottom: 2rem;
    margin-bottom: 4rem;
    z-index: 9200;

    @media (min-width: $size--tablet) {
      left: calc(100vw - 360px);
      max-width: 360px;
      padding: 1rem;
      padding-bottom: 3rem;
    }
    @media (min-width: $size--desktop) {
      left: calc(100vw - 400px);
      max-width: 400px;
    }
  }

  .button--success {
    background: darken($color--success, 5%);
    border: 1px solid darken($color--success, 10%);
    padding: 0.7rem 2rem 0.55rem 2rem;
    color: #fff;

    @media (min-width: $size--tablet) {
      padding: 1rem 2rem 0.8rem 2rem;
    }

    &:hover {
      background: darken($color--success, 15%);
      border-color: darken($color--success, 0%);
      color: #fff;
      box-shadow: none;
    }
  }
}
