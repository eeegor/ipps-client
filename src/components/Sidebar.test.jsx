import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { Sidebar } from './Sidebar';

describe('Sidebar', () => {
  it('renders markup correctly', () => {
    const wrapper = <Sidebar>Hello</Sidebar>;
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle isAuth correctly', () => {
    const wrapper = shallow(<Sidebar isAuth={false}>Hello</Sidebar>);
    expect(wrapper.contains('sidebar__container')).not.toEqual(true);
  });

  it('should handle onLogout event', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <Sidebar isAuth={true} onLogout={callback}>
        Hi
      </Sidebar>
    );
    wrapper
      .find('.button--logout')
      .first()
      .simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
  });

  it('should handle onSetFilterField event', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <Sidebar isAuth={true} onSetFilterField={callback}>
        Hi
      </Sidebar>
    );
    wrapper
      .find('.filter')
      .find('input')
      .first()
      .simulate('change');
    expect(callback).toHaveBeenCalledTimes(1);
  });

  it('should handle onApplyFilter event', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <Sidebar isAuth={true} onApplyFilter={callback}>
        Hi
      </Sidebar>
    );
    wrapper
      .find('.filter__apply')
      .find('button')
      .first()
      .simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
  });
});
