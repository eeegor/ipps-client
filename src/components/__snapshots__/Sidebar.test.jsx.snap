// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders markup correctly 1`] = `
<div
  className="sidebar"
/>
`;
